File: 简单/Java/leetCode/src/main/java/com/kurumi/leetCode/双指针/LeetCode_0344.java
/**
 * @Author: kurumiwj
 * @Date: 2024-10-19 16:59:00
 * @Description: 344. 反转字符串
 */
package com.kurumi.leetCode.双指针;

public class LeetCode_0344 {
    public void reverseString(char[] s) {
        int left = 0, right = s.length - 1;
        while (left < right) {
            s[left] ^= s[right];
            s[right] ^= s[left];
            s[left++] ^= s[right--];
        }
    }
}

File: 卡码/代码随想录/Go/字符串/反转字符串/LeetCode_0344.go
//344. 反转字符串
package main

func reverseString(s []byte)  {
	left,right:=0,len(s)-1
	for left<right {
		s[left]^=s[right]
		s[right]^=s[left]
		s[left]^=s[right]
		left++
		right--
	}
}
